.travis.yml

```diff
@@ -22,16 +27,32 @@ install:
   # Useful for debugging any issues with conda
   - conda info -a
 
-  # Install dependencies, use python 2.7 to test our lower requirements still hold, python3 does not work with scipy 0.14
-  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION libgfortran numpy scipy matplotlib
+  # Install dependencies and enter test environment
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy scipy future nomkl sphinx sphinx_rtd_theme
   - source activate test-environment
+
+  # Sphinx doc integration
+  - pip install travis-sphinx
+
+  # Install our package
   - pip install -e .
+
+
 before_script:
   - pip install -r test_requirements.txt
-  
-# command to run tests
-script: 
+
+script:
+  # Run tests, including PEP8 check, and produce a coverage report
   - py.test --doctest-modules --cov --cov-report term-missing --pep8
-  
+  # Build the Sphinx doc
+  - cd $TRAVIS_BUILD_DIR/doc/source && python generate_doc.py && cd -
+  - travis-sphinx -n -s $TRAVIS_BUILD_DIR/doc/source build
```

Comment:
What is "This"?
Which script?

Anyway, I'm not sure what exactly the `travis-sphinx` command does other than `make html`, but if so, this is the more robust variant.